### DISCUSSION TITLE
include Calibration settings in main sketch

### DISCUSSION THREAD

**Discussion by JKeyRandom** [ImpulseAdventure/GUIslice#363] [Q&A] [answered]
Good morning, could you include the calibration parameters in the main code of the arduino ide? Is there any way to do it?

`#define ADATOUCH_X_MIN    3911
  #define ADATOUCH_X_MAX    171
  #define ADATOUCH_Y_MIN    3937
  #define ADATOUCH_Y_MAX    393
  #define ADATOUCH_REMAP_YX 0`

Thank you very much for the great library.

**Comment by ImpulseAdventure** [comment, marked as answer]
Hi @JosekeyRandom --

Yes, you can define the calibration within your sketch.

Here is an example:
```c++
    gslc_SetTouchRemapCal(&m_gui, nXMin, nXMax, nYMin, nYMax);
    gslc_SetTouchRemapYX(&m_gui, nRemapYX);
    gslc_SetTouchRemapEn(&m_gui, true);
```

**Reply by JKeyRandom** [reply]
@ImpulseAdventure  Ohh! Thank you very much it works perfect, could you do the same with these parameters? 

```
#define ADATOUCH_PRESS_MIN  300
#define ADATOUCH_PRESS_MAX  4000
```

**Reply by ImpulseAdventure** [reply]
Hi @JosekeyRandom -- yes, I should be able to add the ability to control the touch thresholds at run-time too. Let me add this to my to-do list. thanks!

**Reply by ImpulseAdventure** [reply]
Hi @JosekeyRandom -- just letting you know that your additional feature request for enabling runtime configuration of the resistive touch calibration pressure has been integrated.

You can now call the following from your sketch: (sets `PRESS_MIN` and `PRESS_MAX`)
```c++
gslc_SetTouchPressCal(&m_gui,2000,4000);
```

Theoretically, this should also enable the future creation of a touch pressure calibration sketch too.

**Reply by JKeyRandom** [reply]
Thank you very much for the update, I appreciate it very much!!
